com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/variables/VariableProxy.java

```diff
@@ -109,4 +110,12 @@ public boolean isUnboundedType() {
     public void setUnboundedType(boolean isUnboundedType) {
         this.isUnboundedType = isUnboundedType;
     }
+
+    public Integer getReferencedVariableId() {
```

Comment:
I suggest to add a property `isLazyResolved()` instead of `getReferencedVariableId`. Its default value is `false`. And set it to `true` when you return that variable as lazy presentation.